### PR TITLE
Manage communities

### PR DESCRIPTION
**Pull request by b-avb**
- Create space for a community
- Get a space by id
- Upload image for a community

### CODE REVIEW

File: src/bot/routes/rooms.rs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    fields(
+        space_name = %form.name,
+        space_industry = %form.industry
+    )
+)]
+pub async fn create(
+    form: web::Json<CommunityData>,
+    matrix_client: web::Data<MatrixClient>,
+) -> Result<impl Responder, ServerError> {
+    println!("{:?}", form);
```
Comment (olanod) [inline src/bot/routes/rooms.rs:61]: Can we log stuff with the log crate?

File: src/bot/routes/rooms.rs [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+#[tracing::instrument(name = "Create a room", skip(matrix_client))]
+pub async fn on_handle_create_room(
+    room: CommunityData, 
+    matrix_client: &MatrixClient,
+) -> Result<String, Error> {
+    let mut request = api::client::room::create_room::v3::Request::new();
+    request.name = Some(&room.name);
+    
+    let topic = format!("{}\nTAG_VIRTO: {}",
```
Comment (olanod) [inline src/bot/routes/rooms.rs:79]: This topic will be visible in UIs like element no? can the industry be part of a state event in json or similar? or we can give it a "name" that feels less machine readable(e.g. `"🏢️:"`)

File: src/bot/routes/rooms.rs (hunk 1 above)
Comment (b-avb) [reply src/bot/routes/rooms.rs]: It isn't really necessary to print the form, all the info is on the tracing

**Review by olanod — APPROVED** [review]
The only thing I would double check now that the logo is kept in a buffer in memory is if the upload endpoint can have a max upload size or similar to avoid bad users from using up all the memory and crashing the service.

**Pull request merged**